airbyte-server/src/main/java/io/airbyte/server/handlers/ArchiveHandler.java

```diff
@@ -0,0 +1,181 @@
+import io.airbyte.commons.lang.Exceptions;
+import io.airbyte.config.ConfigSchema;
+import io.airbyte.config.DestinationConnection;
+import io.airbyte.config.SourceConnection;
+import io.airbyte.config.StandardDestinationDefinition;
+import io.airbyte.config.StandardSourceDefinition;
+import io.airbyte.config.StandardSync;
+import io.airbyte.config.StandardSyncSchedule;
+import io.airbyte.config.StandardWorkspace;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.config.persistence.PersistenceConstants;
+import io.airbyte.server.converters.ConfigFileArchiver;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArchiveHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ArchiveHandler.class);
+  private static final String ARCHIVE_FILE_NAME = "airbyte_config_data";
+  private static final String VERSION_FILE_NAME = "VERSION";
+
+  private final String version;
+  private final ConfigRepository configRepository;
+
+  public ArchiveHandler(final String version, final ConfigRepository configRepository) {
+    this.version = version;
+    this.configRepository = configRepository;
+  }
+
+  public Path exportData() {
+    try {
+      final Path tempFolder = Files.createTempDirectory("airbyte_archive");
+      final Path archive = Files.createTempFile(ARCHIVE_FILE_NAME, ".tar.gz");;
+      archive.toFile().deleteOnExit();
+      try {
+        exportVersionFile(tempFolder);
+        exportAirbyteConfig(tempFolder);
+        exportAirbyteDatabase(tempFolder);
+        Archives.createArchive(tempFolder, archive);
+      } finally {
+        FileUtils.deleteDirectory(tempFolder.toFile());
+      }
+      return archive;
+    } catch (IOException e) {
+      LOGGER.error("Export Data failed.");
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void exportVersionFile(Path tempFolder) throws IOException {
+    final File versionFile = Files.createFile(tempFolder.resolve(VERSION_FILE_NAME)).toFile();
+    FileUtils.writeStringToFile(versionFile, version, Charset.defaultCharset());
+  }
+
+  private void exportAirbyteConfig(Path tempFolder) {
+    LOGGER.info("Exporting Airbyte Configs");
+    final ConfigFileArchiver configFileArchiver = new ConfigFileArchiver(version, tempFolder);
+    Exceptions.toRuntime(() -> {
+      configFileArchiver.writeConfigsToArchive(ConfigSchema.STANDARD_WORKSPACE,
+          List.of(configRepository.getStandardWorkspace(PersistenceConstants.DEFAULT_WORKSPACE_ID)));
+      configFileArchiver.writeConfigsToArchive(ConfigSchema.STANDARD_SOURCE_DEFINITION, configRepository.listStandardSources());
```

Comment:
I know it is not part of the PR but the method naming scheme is inconsistent between models. (plural & full name).